### PR TITLE
Extract escapeCSVValue util to its own module

### PR DESCRIPTION
**Pull request by acelaya**
The `escapeCSVValue` utility was implemented as part of the CSV export annotations.

This PR extracts it so that it's easier to test independently, and reuse if needed.

### CODE REVIEW

File: src/shared/csv.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Escapes provided value so that it can be used inside a CSV without breaking
+ * its format.
+ * It checks if the value contains a comma, newline or double quote, then wraps
+ * it in double quotes.
+ * It also escapes existing double quotes.
+ */
+export function escapeCSVValue(value: string): string {
```
Comment (robertknight) [inline src/shared/csv.ts:8]: This can be shortened to something like "Escape a CSV field value", along with a short example and any spec this follows (eg. https://www.ietf.org/rfc/rfc4180.txt)

The code and tests provide additional details if the reader needs them.

File: src/shared/csv.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  if (/[",\n]/.test(value)) {
```
Comment (robertknight) [inline src/shared/csv.ts:9]: Based on the linked spec, `\r` should trigger an escape as well.

**Review by robertknight — COMMENTED** [review]
This looks mostly good, but the comment can be simplified to avoid duplicating the code/tests. Also as an edge case, it looks like `\r` on its own should be escaped.

File: src/shared/csv.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Escape a CSV field value (see https://www.ietf.org/rfc/rfc4180.txt)
+ *
+ * - foo -> foo
+ * - foo,bar -> "foo,bar"
+ * - with "quoted" text -> "with ""quoted"" text"
+ */
```
Comment (robertknight) [inline src/shared/csv.ts:7]: Much better 👍

**Review by robertknight — APPROVED** [review]
